src/Google/NotificationsService.php

```diff
@@ -107,4 +141,74 @@ protected function do_request( $args ) {
 	public function get_route(): string {
 		return $this->route;
 	}
+
+	/**
+	 * If the Notifications are enabled
+	 *
+	 * @return bool
+	 */
+	public function is_enabled(): bool {
+		return apply_filters( 'woocommerce_gla_notifications_enabled', true );
+	}
+
+	/**
+	 * Find a product matching the provided status
+	 *
+	 * @param int    $product_id The product Id to filter
+	 * @param string $topic The product topic to filter
+	 *
+	 * @return int|null
+	 */
+	public function filter_product( $product_id, $topic ) {
```

Comment:
Hi @mikkamp. The only reason for this function is to allow the user to hook a filter (as we discussed) 

The function calls `$this->product_repository->find_notification_products` that internally query by ids and calls `execute_woocommerce_query` and `prepare_query_args`. This is how we allow the user to filter the query args i sync strategy. 

If this is not needed I agree to remove this function.

> We need to set a NotificationStatus as a product meta.

The reason of this metas is to know if a product notified `product.create`  or other notifications so we can handle the updates/creates/deletes.

Reply:
I remove tho the `filter_product` function. I would appreciate some indications about where to include the filter for the users.

Reply:
Sorry for any confusing. When I said we'd like to retain a filter my intention was to still allow overriding when a notification gets sent through a WP hook (filter). The querying and filtering of only eligible products for notifications isn't what I meant to retain.

As for where to place the hook, I believe this should be called twice:
1. Just before being scheduled
2. Just before the scheduled action is run (in case the product changed in the meantime)

Right now you already have `is_ready_to_notify` which gets checked before scheduling, I think that's the right place to add a hook, and also make sure it checks all the conditions like:
- Channel Visibility is set to sync
- it has the right status for a notification (published)
- is a syncable product
- is the right product type
- call filter at the end so extensions (or our compatibility code) can prevent this notification from being scheduled/sent

We'd just then need to repeat the call when the action is run and if `is_ready_to_notify` returns false it would prevent the notification from being sent.

Reply:
Hi @mikkamp thanks for the insightful clarifications. I added some useful hooks here https://github.com/woocommerce/google-listings-and-ads/pull/2202/commits/99f5b9ede4459201090e16b51d6a790add75a03b